src/PointwiseFunctions/AnalyticSolutions/GrMhd/BondiMichelMHD.hpp

```diff
@@ -0,0 +1,207 @@
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/Interpolation/BarycentricRational.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/MakeArray.hpp"            // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace Solutions {
+
+/*!
+ * \brief Bondi-Michel accretion with superposed magnetic field in Schwarzchild
+ * spacetime.
+ *
+ * Modulo the magnetic field, an analytic solution to the 3-D GrMhd system.
+ * The user specifies the sonic radius \f$r_c\f$ and sonic rest mass density
+ * \f$\rho_c\f$, which are the radius and rest mass density at the sonic point,
+ * the radius at which the fluid's eulerian velocity as seen by a distant
+ * observer overtakes its sound speed. These quantities can be related to the
+ * sound speed at infinity. In the case of the interstellar medium, the sound
+ * speed is \f$\approx 10^{-4}\f$, which results in a sonic radius of
+ * \f$\approx 10^8 M\f$ for \f$\Gamma \neq 5/3\f$ (Rezzola and Zanotti, 2013).
+ *
+ * The user also specifies the max radius, which is useful if one does not want
+ * to extend the domain up to the remote sonic point. Additionally specified by
+ * the user are the adiabatic exponent \f$\Gamma\f$, and the magnitude of the
+ * magnetic field \f$B\f$.
+ * In Schwarschild-Cartesian coordinates \f$(x, y, z)\f$, where
+ * \f$ r = \sqrt{x^2 + y^2 + z^2}\f$, the superposed magnetic field is
+ *
+ * \f{align*}
+ * B_x(\vec{x},t) &= \frac{B x}{r^3 \sqrt{1 + 2/r}} \\
+ * B_y(\vec{x},t) &= \frac{B y}{r^3 \sqrt{1 + 2/r}} \\
+ * B_z(\vec{x},t) &= \frac{B z}{r^2 \sqrt{1 + 2/r}}
+ * \f}
+ */
+class BondiMichelMHD {
+ public:
+  /// The radius at which the fluid becomes supersonic.
+  struct SonicRadius {
+    using type = double;
+    static constexpr OptionString help = {
+        "The radius at which the fluid becomes supersonic."};
+    static type lower_bound() { return 10; }
+  };
+
+  /// The radius of the computational domain,
+  /// beyond which the solution is not used.
+  struct MaxRadius {
+    using type = double;
+    static constexpr OptionString help = {
+        "The radius beyond which the solution is not used."};
+  };
+
+  /// The density of the fluid at the sonic radius.
+  struct SonicRhoZero {
+    using type = double;
+    static constexpr OptionString help = {
+        "The density of the fluid at the sonic radius."};
+    static type lower_bound() { return 0.0; }
+  };
+
+  /// The adiabatic exponent for the polytropic fluid.
+  struct AdiabaticExponent {
+    using type = double;
+    static constexpr OptionString help = {
+        "The adiabatic exponent for the polytropic fluid."};
+    static type lower_bound() { return 1.0; }
+  };
+
+  /// The strength of the radial magnetic field.
+  struct MagneticField {
```

Comment:
Could you find another name depending on what B is (see question above)? Let's use "magnetic field" for the actual field only.